help_internal_test.go

```diff
@@ -97,3 +130,94 @@ func TestHelpWithHiddenActions(t *testing.T) {
 
 	assert.Equal(t, "🤝 Hi, `Daniel Quinn`! I'm `robert` (engine `v1.0.0`) and I listen to the team's chat and provides automated functions :genie:.\n", a.Text)
 }
+
+
+
+func TestHelpWithNamespacingEnabledWithCommandOption(t *testing.T) {
+	s, err := New("robert", config.NewViperWithDefaults())
+	s.RegisterPlugin(newPluginWithActionsOfAllTypes(false))
+	s.cmdMatcher = NewTestCmdMatcher("")
+
+	require.NoError(t, err)
+
+	help := s.newHelpPlugin("1.0.0")
+	help.UserInfoFinder = &userInfoFinder{}
+
+	cmd := help.Commands[0]
+	assert.False(t, cmd.Match(&IncomingMessage{NormalizedText: " help"}))
+	require.True(t, cmd.Match(&IncomingMessage{NormalizedText: "help"}))
+	assert.True(t, cmd.Match(&IncomingMessage{NormalizedText: "help and something else"}))
+
+	a := cmd.Answer(&IncomingMessage{NormalizedText: "help"})
+	require.NotNil(t, a)
+
+	assert.Equal(t, "🤝 Hi, `Daniel Quinn`! I'm `robert` (engine `v1.0.0`) and I listen to the team's chat and provides automated functions :genie:.\n\n"+
+		"I currently support the following commands:\n\t• `thank <someone of something to thank>` - Format a thank you note\n\nAnd listen for the following:\n"+
+		"\t• `say `chickadee` and hear a chirp` - Chirp when hearing people talk about chickadees\n\nAnd do those things periodically:\n"+
+		"\t• [`thank`] `Every 30 seconds` (`Local`) - Sends a heartbeat every 30 seconds\n", a.Text)
+}
+
+func TestHelpWithNamespacingEnabledWithCommandOption2(t *testing.T) {
+	s, err := New("robert", config.NewViperWithDefaults())
+	s.RegisterPlugin(newPluginWithActionsOfAllTypes(false))
+	s.cmdMatcher = NewTestCmdMatcher("!!")
+
+	require.NoError(t, err)
+
+	help := s.newHelpPlugin("1.0.0")
+	help.UserInfoFinder = &userInfoFinder{}
+
+	cmd := help.Commands[0]
+	assert.False(t, cmd.Match(&IncomingMessage{NormalizedText: " help"}))
+	require.True(t, cmd.Match(&IncomingMessage{NormalizedText: "help"}))
+	assert.True(t, cmd.Match(&IncomingMessage{NormalizedText: "help and something else"}))
+
+	a := cmd.Answer(&IncomingMessage{NormalizedText: "help"})
+	require.NotNil(t, a)
+
+	assert.Equal(t, "🤝 Hi, `Daniel Quinn`! I'm `robert` (engine `v1.0.0`) and I listen to the team's chat and provides automated functions :genie:.\n\n"+
+		"I currently support the following commands:\n\t• `!!thank <someone of something to thank>` - Format a thank you note\n\nAnd listen for the following:\n"+
+		"\t• `say `chickadee` and hear a chirp` - Chirp when hearing people talk about chickadees\n\nAnd do those things periodically:\n"+
+		"\t• [`thank`] `Every 30 seconds` (`Local`) - Sends a heartbeat every 30 seconds\n", a.Text)
+}
+
+func TestHelpWithNamespacingDisabledWithCommandOption(t *testing.T) {
```

Comment:
Can you rename one of the two test functions to avoid using a numeric suffix? What do you think of naming this one `TestHelpWithNamespacingDisabledWithBlankPrefixCommandOption`?

Reply:
Fixed in 9d4e0806f32c5e5f525bb5b708a20db561d4017e